src/libPersistence/BlockStorage.cpp

```diff
@@ -217,15 +244,15 @@ bool BlockStorage::PutTempState(const unordered_map<Address, Account>& states) {
     states_str.emplace(state.first.hex(),
                        DataConversion::CharArrayToString(rawBytes));
   }
-
+  unique_lock<shared_timed_mutex> g(m_mutexTempState);
   return m_tempStateDB->BatchInsert(states_str);
 }
 
 bool BlockStorage::GetTempStateInBatch(leveldb::Iterator*& iter,
                                        vector<StateSharedPtr>& states) {
   // LOG_MARKER();
 
-  lock_guard<mutex> g(m_mutexTempState);
+  shared_lock<shared_timed_mutex> g(m_mutexTempState);
 
   if (iter == nullptr) {
     iter = m_tempStateDB->GetDB()->NewIterator(leveldb::ReadOptions());
```

Comment:
Maybe let `m_mutexTempState` to protect `iter = m_tempStateDB->GetDB()->NewIterator(leveldb::ReadOptions());` only

Reply:
cannot do that...the iterator is keep changing in the for loop